website/src/pages/docs/features/apollo-federation.mdx

```diff
@@ -28,44 +28,187 @@ we can use GraphQL Yoga to build our gateway server.
+})
+```
+
+> You can also check a full example on our GitHub repository
+> [here](https://github.com/dotansimha/graphql-yoga/tree/main/examples/apollo-managed-federation)
+
+#### Configuration
 
+By default, the plugin will use standard Apollo GraphOS environment variables.
+
+```bash
+APOLLO_KEY='<YOU_APOLL_API_KEY>' APOLLO_GRAPH_REF='<YOUR_GRAPH_ID>[@<VARIANT>]' node src/index.mjs
+```
+
+If you don't know where to find these values, please refer to the
+[Apollo GraphOS documentation](https://www.apollographql.com/docs/federation/managed-federation/setup#4-connect-your-router-to-graphos).
+
+You can also provide this configuration programaitically by providing a configurtion object to the
+plugin, along with other options to customize the polling/retry behavior.
+
+```js
 const yoga = createYoga({
   plugins: [
-    useApolloFederation({
-      gateway
+    useManagedFederation({
+      apiKey: '<YOU_APOLL_API_KEY>',
+      graphRef: '<YOUR_GRAPH_ID>[@<VARIANT>]',
+      maxRetries: 5, // max retries in case of loading failure
+      retryDelaySeconds: 0, // delay between retries
+      minDelaySeconds: 1 // minimum delay between polling requests
     })
   ]
 })
+```
 
-// Start the server and explore http://localhost:4000/graphql
+Other more advanced options are available, please refer to the type definitions for more
+information.
+
+### Using the file system
+
+To create a gateway server withtout using Apollo GraphOS, you can use the
+`@graphql-tools/federation` which can generate an executable schema from a supergraph SDL.
+
+#### Installation
+
+```sh npm2yarn
+npm i graphql-yoga graphql @graphql-tools/federation
+```
+
+#### Example
+
+To generate the schema, we first need the supergraph SDL. This can be generated by the Apollo Rover
+CLI or fetched from a schema registry (like Hive or Apollo GraphOS).
+
+To generate the supergraph SDL locally, you first have to define a supergraph configuration by
+creating a `supergraph.yaml` file.
+
+```yaml filename="supergraph.yaml"
+subgraphs:
+  accounts:
+    routing_url: http://localhost:4001/graphql
+    schema:
+      file: ./services/accounts.graphql
+  inventory:
+    routing_url: http://localhost:4002/graphql
+    schema:
+      file: ./services/inventory.graphql
+```
+
+You can then use the Apollo Rover CLI to generate the supergraph SDL.
+
+```bash
+rover supergraph compose --config ./supergraph.yaml > supergraph.graphql
+```
+
+You can now serve the gateway using GraphQL Yoga.
+
+```ts filename="src/index.mjs"
+import { readFileSync } from 'fs'
+import { createServer } from 'node:http'
+import { createYoga } from 'graphql-yoga'
+import { getStitchedSchemaFromSupergraphSdl } from '@graphql-tools/federation'
+
+// This doesn't have to be from a file system, it can be fetched via HTTP from a schema registry
+const supergraphSdl = readFileSync('./supergraph.graphql').toString()
+const schema = await getStitchedSchemaFromSupergraphSdl({ supergraphSdl })
+const yoga = createYoga({ schema })
```

Comment:
```suggestion
const yoga = createYoga({ 
 schema: getStitchedSchemaFromSupergraphSdl({
    supergraphSdl: readFileSync('./supergraph.graphql', 'utf-8')
 })
})
```

Reply:
`getStitchedSchemaFromSupergraphSdl` is not Promise so we don't need an `await` there.

Reply:
We should update the documentation of stitching then, I copy pasted it from there